Custom good names

After change in pull request https://github.com/PyCQA/pylint-django/pull/109 seems to be broken to ability to set own good names. Example pylintrc:

```
[MASTER]
load-plugins=pylint_django

[BASIC]
good-names=_,dt,e,f,default_app_config,logger
```

In version v0.7.2 it was working properly but starting version 0.8.0 it can be one or other. Probably previous version was merging two lists of good names which is not the case now.

I have the same issue that when after setting `good-names` in `.pylintrc`, false positive errors are raised for defaults of `pylint-django`. I have investigated the problem and it is because by setting `good-names` in `.pylintrc`, the defaults together with `pylint-django` good names are overridden. This can be easily proven:
1.  Execute pylint outside project with rcfile set - false positive error is raised for urlpatterns:
```
$ pylint --load-plugins=pylint_django --rcfile DJANGO_PROJECT_DIR/.pylintrc  DJANGO_PROJECT_DIR/MYPROJECT/MYPROJECT/urls.py
DJANGO_PROJECT_DIR/MYPROJECT/MYPROJECT/urls.py:26:0: C0103: Constant name "urlpatterns" doesn't conform to UPPER_CASE naming style (invalid-name)
```
2. Execute pylint outside project without rcfile set - false positive error is not raised:
```
$ pylint --load-plugins=pylint_django DJANGO_PROJECT_DIR/MYPROJECT/MYPROJECT/urls.py
# urlpatterns issue is not reported
```

The problem is that pylint-django is extending **default** values:
https://github.com/PyCQA/pylint-django/blob/c5b5bfdef66453575074b36018ee716411bcb0a4/pylint_django/plugin.py#L19
which **pylint overrides with values from configuration file** [1]:
```python
        if config_parser.has_option("MASTER", "load-plugins"):
            plugins = utils._splitstrip(config_parser.get("MASTER", "load-plugins"))
            linter.load_plugin_modules(plugins)
        # now we can load file config and command line, plugins (which can
        # provide options) have been registered
        linter.load_config_file()
```

Hence, I suggest that pylint plugins should have hooks which are called after configuration is loaded to inject plugin configurations. But this decition should be done in mainly in pylint.
[1] https://github.com/PyCQA/pylint/blob/fcc01516ae176ad3fdedc4497328105f3314e376/pylint/lint.py#L1566-L1571

--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -9,3 +9,4 @@
 * [atodorov](https://github.com/atodorov)
 * [bittner](https://github.com/bittner)
 * [federicobond](https://github.com/federicobond)
+* [matusvalo](https://github.com/matusvalo)

--- a/pylint_django/checkers/db_performance.py
+++ b/pylint_django/checkers/db_performance.py
@@ -13,6 +13,7 @@ from pylint import interfaces
 from pylint import checkers
 from pylint.checkers import utils
 from pylint_django.__pkginfo__ import BASE_ID
+from pylint_django import compat
 
 
 def _is_addfield_with_default(call):
@@ -113,12 +114,16 @@ class NewDbFieldWithDefaultChecker(checkers.BaseChecker):
                     self.add_message('new-db-field-with-default', args=module.name, node=node)
 
 
-def register(linter):
-    """Required method to auto register this checker."""
+def load_configuration(linter):
     # don't blacklist migrations for this checker
     new_black_list = list(linter.config.black_list)
     if 'migrations' in new_black_list:
         new_black_list.remove('migrations')
     linter.config.black_list = new_black_list
 
+
+def register(linter):
+    """Required method to auto register this checker."""
     linter.register_checker(NewDbFieldWithDefaultChecker(linter))
+    if not compat.LOAD_CONFIGURATION_SUPPORTED:
+        load_configuration(linter)

--- a/pylint_django/compat.py
+++ b/pylint_django/compat.py
@@ -22,3 +22,7 @@ except ImportError:
     except ImportError:
         from astroid.util import Uninferable
 
+import pylint
+
+# pylint before version 2.3 does not support load_configuration() hook.
+LOAD_CONFIGURATION_SUPPORTED = pylint.__pkginfo__.numversion >= (2, 3)

--- a/pylint_django/plugin.py
+++ b/pylint_django/plugin.py
@@ -7,13 +7,12 @@ from pylint_django.checkers import register_checkers
 # we want to import the transforms to make sure they get added to the astroid manager,
 # however we don't actually access them directly, so we'll disable the warning
 from pylint_django import transforms  # noqa, pylint: disable=unused-import
+from pylint_django import compat
 
 
-def register(linter):
+def load_configuration(linter):
     """
-    Registering additional checkers.
-
-    However, we will also use it to amend existing checker config.
+    Amend existing checker config.
     """
     name_checker = get_checker(linter, NameChecker)
     name_checker.config.good_names += ('qs', 'urlpatterns', 'register', 'app_name', 'handler500')
@@ -21,6 +20,11 @@ def register(linter):
     # we don't care about South migrations
     linter.config.black_list += ('migrations', 'south_migrations')
 
+
+def register(linter):
+    """
+    Registering additional checkers.
+    """
     # add all of the checkers
     register_checkers(linter)
 
@@ -32,3 +36,6 @@ def register(linter):
         # probably trying to execute pylint_django when Django isn't installed
         # in this case the django-not-installed checker will kick-in
         pass
+
+    if not compat.LOAD_CONFIGURATION_SUPPORTED:
+        load_configuration(linter)

